### ISSUE TITLE
Fail to install gnutls in CentOS6

### ISSUE THREAD

**Issue by Gwang-Jin** [Linuxbrew/homebrew-core#7886]
Hello. When I try to install `gnutils`, I met an error message:
````
libtool: compile:  gcc-5 -DHAVE_CONFIG_H -I. -I../.. -I./../../gl -I./../../gl -I./../includes -I./../includes -I./.. -Wtype-limits -fno-common -W -Wabi -Waddress -Waggressive-loop-optimizations -Wall -Wattributes -Wbad-function-cast -Wbool-compare -Wbuiltin-macro-redefined -Wcast-align -Wchar-subscripts -Wchkp -Wclobbered -Wcomment -Wcomments -Wcoverage-mismatch -Wcpp -Wdate-time -Wdeprecated -Wdeprecated-declarations -Wdesignated-init -Wdisabled-optimization -Wdiscarded-array-qualifiers -Wdiscarded-qualifiers -Wdiv-by-zero -Wdouble-promotion -Wempty-body -Wendif-labels -Wenum-compare -Wextra -Wformat-contains-nul -Wformat-extra-args -Wformat-security -Wformat-zero-length -Wfree-nonheap-object -Wignored-qualifiers -Wimplicit -Wimplicit-function-declaration -Wimplicit-int -Wincompatible-pointer-types -Winit-self -Wint-conversion -Wint-to-pointer-cast -Winvalid-memory-model -Winvalid-pch -Wjump-misses-init -Wlogical-not-parentheses -Wlogical-op -Wmain -Wmaybe-uninitialized -Wmemset-transposed-args -Wmissing-braces -Wmissing-declarations -Wmissing-field-initializers -Wmissing-include-dirs -Wmissing-parameter-type -Wmissing-prototypes -Wmultichar -Wnarrowing -Wnested-externs -Wnonnull -Wodr -Wold-style-declaration -Wold-style-definition -Wopenmp-simd -Woverflow -Woverride-init -Wpacked -Wpacked-bitfield-compat -Wparentheses -Wpointer-arith -Wpointer-sign -Wpointer-to-int-cast -Wpragmas -Wreturn-local-addr -Wreturn-type -Wsequence-point -Wshadow -Wshift-count-negative -Wshift-count-overflow -Wsizeof-array-argument -Wsizeof-pointer-memaccess -Wstrict-aliasing -Wstrict-prototypes -Wsuggest-attribute=format -Wsuggest-final-methods -Wsuggest-final-types -Wswitch -Wswitch-bool -Wsync-nand -Wtrampolines -Wtrigraphs -Wtype-limits -Wuninitialized -Wunknown-pragmas -Wunused -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-function -Wunused-label -Wunused-local-typedefs -Wunused-macros -Wunused-parameter -Wunused-result -Wunused-value -Wunused-variable -Wvarargs -Wvariadic-macros -Wvector-operation-performance -Wvolatile-register-var -Wwrite-strings -Warray-bounds=2 -Wnormalized=nfc -Wno-missing-field-initializers -Wno-unused-parameter -fdiagnostics-show-option -I/home/skpark/.linuxbrew/Cellar/nettle/3.4/include -I/home/skpark/.linuxbrew/Cellar/libtasn1/4.13/include -I/home/skpark/.linuxbrew/Cellar/p11-kit/0.23.11/include/p11-kit-1 -g -O2 -c x509_dn.c  -fPIC -DPIC -o .libs/x509_dn.o
In file included from ./../pkcs11_int.h:34:0,
                 from x509.c:37:
/home/skpark/.linuxbrew/Cellar/p11-kit/0.23.11/include/p11-kit-1/p11-kit/uri.h:100:68: error: unknown type name 'CK_INFO'
                                                              const CK_INFO *info);
                                                                    ^
/home/skpark/.linuxbrew/Cellar/p11-kit/0.23.11/include/p11-kit-1/p11-kit/uri.h:105:68: error: unknown type name 'CK_SLOT_INFO'
                                                              const CK_SLOT_INFO *slot_info);
                                                                    ^
/home/skpark/.linuxbrew/Cellar/p11-kit/0.23.11/include/p11-kit-1/p11-kit/uri.h:114:68: error: unknown type name 'CK_TOKEN_INFO'
                                                              const CK_TOKEN_INFO *token_info);
                                                                    ^
/home/skpark/.linuxbrew/Cellar/p11-kit/0.23.11/include/p11-kit-1/p11-kit/uri.h:135:68: error: unknown type name 'CK_ATTRIBUTE'
                                                              const CK_ATTRIBUTE *attrs,
                                                                    ^
make[3]: *** [x509.lo] Error 1
````

But I found `CK_SLOT_INFO` was defined in `p11-kit/pkcs11.h` which was included in `uri.h`. I am searching for the reason. Thank you.

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
[https://gist.github.com/7a217ea761da29bef28932500858d7bf](https://gist.github.com/7a217ea761da29bef28932500858d7bf)

To help us debug your issue please explain:
- What you were trying to do (and why)
- What happened (include command output)
- What you expected to happen
- Step-by-step reproduction instructions (by running `brew install` commands)

**Comment by rcoacci** [comment]
I fixed it in CentOS 7 adding `--without-p11-kit` to `brew install gnutls`.

**Comment by iMichka** [comment]
This is an upstream bug: https://github.com/p11-glue/p11-kit/issues/159

Seems it is fixed by 0.23.12: https://github.com/p11-glue/p11-kit/pull/160
